### PR TITLE
Replacing skydns with kubedns for the juju cluster. #29720

### PR DESCRIPTION
**Pull request by mbruzek**
``` release-note
* Updating the cluster/juju provider to use kubedns in place of skydns.
```

### CODE REVIEW

**Comment by mbruzek** [comment]
@thockin Is this what you are looking for. We used the .base templates and a sed script. Our ops code had to change to use the new file names.

**Comment by thockin** [comment]
ok to test

File: cluster/juju/layers/kubernetes/templates/kubedns-rc.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+# limitations under the License.
+
+# This file should be kept in sync with cluster/images/hyperkube/dns-rc.yaml
+
+# Warning: This is a file generated from the base underscore template file: skydns-rc.yaml.base
+
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: kube-dns-v18
```
Comment (thockin) [inline cluster/juju/layers/kubernetes/templates/kubedns-rc.yaml]: Is there a way to NOT copy the file?  This is just going to drift again.  We usually grep for things, but sometimes things get missed or forgotten.

Comment (mbruzek) [inline cluster/juju/layers/kubernetes/templates/kubedns-rc.yaml]: @thockin We are working toward a solution that would copy the *.base files. The current code renders templates at deploy time and we insert the architecture that the master is running on. So our template needs an architecture variable. I asked you about this on slack and we would have to address all calling cases which is a much larger pull request that is outside the scope of getting current with kubedns. 

Is this change OK to review?

Comment (mbruzek) [inline cluster/juju/layers/kubernetes/templates/kubedns-rc.yaml]: I created  #30017 to track that issue.

**Comment by mbruzek** [comment]
@thockin I found a bug in the original PR. Make a one line change to the original PR. Please let me know if this is OK to review.

**Comment by thockin** [comment]
As long as we get a followup for the real fix, I'll LGTM this.  It's guaranteed to be out of date before long...

**Pull request merged**